parachain/runtime/interlay/src/lib.rs

```diff
@@ -1417,6 +1417,19 @@ impl_runtime_apis! {
         }
     }
 
+    impl module_reward_rpc_runtime_api::RewardApi<
+        Block,
+        VaultId,
+        CurrencyId,
+        Balance
+    > for Runtime {
+        fn compute_reward(account_id: VaultId, currency_id: CurrencyId) -> Result<BalanceWrapper<Balance>, DispatchError> {
+            let result = VaultRewards::compute_reward(currency_id, &account_id)?;
+            let balance = BalanceWrapper::<Balance> { amount: result.try_into().map_err(|_| DispatchError::from(reward::Error::<Runtime, reward::Instance2>::TryIntoIntError))? };
```

Comment:
`reward::Pallet` implements the `Rewards` trait which provides `compute_reward() -> Balance` (and handles the try_into already), but I couldn't figure out a way to call `Rewards::compute_reward()` as opposed to `Pallet::compute_reward()` here. Hence replicating the try_into, which is very verbose and would be great if I could avoid it. Any tips?

Reply:
Needed to cast `VaultRewards` as `Rewards` in order to re-use those methods - see [this commit](https://github.com/interlay/interbtc/pull/705/commits/c319cc33d16427cf1339ba37e2f0b49f49b7b536).